cli/src/lib.rs

```diff
@@ -242,7 +250,9 @@ where
             Arc::clone(&notify_shutdown),
         );
 
-        Self::start_listening_signal(notify_shutdown)?;
+        Self::start_listening_signal(Arc::clone(&notify_shutdown))?;
```

Comment:
Sounds a bit confusing to me, maybe we should rename it to something like `listen_for_signals`?

Reply:
> It's not related to the task

Sure, that's just an observation, I approved regardless.